### PR TITLE
Additional check for ttl < 10s and Consul

### LINKED ISSUE #838: Discovery broken with low TTL
On master, this works:

```
go run main.go join --ttl 10s --heartbeat 1s --addr 127.0.0.1:4242 consul://localhost:8500/test
INFO[0000] Registering on the discovery service every 1s...  addr=127.0.0.1:4242 discovery=consul://localhost:8500/test
```

However this fails mysteriously:

```
go run main.go join --ttl 2s --heartbeat 1s --addr 127.0.0.1:4242 consul://localhost:8500/test
INFO[0000] Registering on the discovery service every 1s...  addr=127.0.0.1:4242 discovery=consul://localhost:8500/test
ERRO[0000] Unexpected response code: 400 (Missing session)
```

### PR DESCRIPTION
**Pull request by abronan**
Fixes #838 

Signed-off-by: Alexandre Beslic <abronan@docker.com>

### CODE REVIEW

**Comment by abronan** [comment]
ping @aluzzardi

File: pkg/store/consul.go [hunk 1]
```diff
@@ -81,8 +85,11 @@ func (s *Consul) setTimeout(time time.Duration) {
 }
 
 // SetEphemeralTTL sets the ttl for ephemeral nodes
-func (s *Consul) setEphemeralTTL(time time.Duration) {
-	s.ephemeralTTL = time
+func (s *Consul) setEphemeralTTL(ttl time.Duration) {
+	if ttl < MinimumTimeToLive {
+		log.Fatal("Consul does not allow a ttl < 10s, please specify a ttl >= 10s")
```
Comment (aluzzardi) [inline pkg/store/consul.go]: I believe a library should never Fatal - we should return an error in initialize

Comment (abronan) [inline pkg/store/consul.go]: Ok, almost forgot that store is in its own package now..

File: pkg/store/consul.go [hunk 2]
```diff
@@ -52,7 +56,10 @@ func InitializeConsul(endpoints []string, options *Config) (Store, error) {
 			s.setTimeout(options.ConnectionTimeout)
 		}
 		if options.EphemeralTTL != 0 {
-			s.setEphemeralTTL(options.EphemeralTTL)
+			err := s.setEphemeralTTL(options.EphemeralTTL)
```
Comment (vieux) [inline pkg/store/consul.go]: nit: can be merged into a single line

File: pkg/store/consul.go [hunk 3]
```diff
@@ -81,8 +88,12 @@ func (s *Consul) setTimeout(time time.Duration) {
 }
 
 // SetEphemeralTTL sets the ttl for ephemeral nodes
-func (s *Consul) setEphemeralTTL(time time.Duration) {
-	s.ephemeralTTL = time
+func (s *Consul) setEphemeralTTL(ttl time.Duration) error {
```
Comment (vieux) [inline pkg/store/consul.go]: can we add a test for `setEphemeralTTL`

File: discovery/kv/kv.go [hunk 4]
```diff
@@ -61,7 +61,15 @@ func (s *Discovery) Initialize(uris string, heartbeat time.Duration, ttl time.Du
 			EphemeralTTL: s.ttl,
 		},
 	)
-	return err
+
+	if err != nil {
+		if err == store.ErrInvalidTTL {
```
Comment (aluzzardi) [inline discovery/kv/kv.go]: Is it really necessary to check this?

AFAIK `CreateStore` never fails unless:
- The backend doesn't exist
- The TTL is invalid

Both should just fail

File: pkg/store/consul.go (hunk 3 above)
Comment (abronan) [inline pkg/store/consul.go]: Don't know where to put the test for now to be honest as we only rely on mocking with `pkg/store`.

Comment (vieux) [inline pkg/store/consul.go]: ok

**Comment by vieux** [comment]
LGTM

**Comment by aluzzardi** [comment]
LGTM

**Pull request merged**